[Cocoa] Move WebNSAttributedStringExtras from /mac to /cocoa

brentfulgham opened the pull request:
#### fb8f23c2355d0dbc7826d7dfb00503d3153a3e1d
<pre>
[Cocoa] Move WebNSAttributedStringExtras from /mac to /cocoa
<a href="https://bugs.webkit.org/show_bug.cgi?id=277718">https://bugs.webkit.org/show_bug.cgi?id=277718</a>
&lt;<a href="https://rdar.apple.com/problem/133333907">rdar://problem/133333907</a>&gt;

Reviewed by Aditya Keerthi.

The WebNSAttributedStringExtras.mm and WebNSAttributedStringExtras.h files are built
on all Cocoa targets, so hosting them in a Mac-specific subdirectory is misleading.

We also don&apos;t need to forward declare NSAttachmentCharacter, since this is API on all
platforms.

* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/cocoa/WebNSAttributedStringExtras.h: Renamed from Source/WebCore/platform/mac/WebNSAttributedStringExtras.h.
* Source/WebCore/platform/cocoa/WebNSAttributedStringExtras.mm: Renamed from Source/WebCore/platform/mac/WebNSAttributedStringExtras.mm.
(WebCore::attributedStringByStrippingAttachmentCharacters):
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.mm:

Canonical link: <a href="https://commits.webkit.org/282003@main">https://commits.webkit.org/282003@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c788e623a78a44b865e9b48c7181942d751964b7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61599 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40949 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14182 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65569 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12140 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63717 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48633 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12413 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49701 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8424 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Running run-layout-tests-in-stress-mode; Passed layout tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64667 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38056 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53355 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30535 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34713 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10586 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11071 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56507 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10886 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67298 "") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5538 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10642 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/67298 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5564 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53305 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/67298 "") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4554 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9294 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36751 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37836 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38930 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37581 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Looks like a unified sources build issue, that should be fixed by adding `import "ApplePayButtonCocoa.h"` in `ControlFactoryCocoa.mm`.

brentfulgham committed:
[Cocoa] Move WebNSAttributedStringExtras from /mac to /cocoa
https://bugs.webkit.org/show_bug.cgi?id=277718
<rdar://problem/133333907>

Reviewed by Aditya Keerthi.

The WebNSAttributedStringExtras.mm and WebNSAttributedStringExtras.h files are built
on all Cocoa targets, so hosting them in a Mac-specific subdirectory is misleading.

We also don't need to forward declare NSAttachmentCharacter, since this is API on all
platforms.

* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/cocoa/WebNSAttributedStringExtras.h: Renamed from Source/WebCore/platform/mac/WebNSAttributedStringExtras.h.
* Source/WebCore/platform/cocoa/WebNSAttributedStringExtras.mm: Renamed from Source/WebCore/platform/mac/WebNSAttributedStringExtras.mm.
(WebCore::attributedStringByStrippingAttachmentCharacters):
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.mm:

Canonical link: https://commits.webkit.org/282003@main
